### DISCUSSION TITLE
3rd party wildcard certificate possible (instead of Lets Encrypt)?

### DISCUSSION THREAD

**Discussion by mjmare** [caprover/caprover#1743] [Q&A] [answered]
Because I cannot expose port 80 to the internet, Letsencrypt certificate renewal is a problem. For this I have been provided with a wildcard certificate (.crt and .key file) for the domain used. 
I have read https://caprover.com/docs/nginx-customization.html and know how to get the certificate file in the directory /captain/data/nginx-shared, mounted by default. 
However, I'm a bit at a loss on how to set the wildcard cert for all the whole domain (or even a subdomain).

I am under the impression that the docs are not quite correct: the files etc/nginx/nginx.conf, /etc/nginx/conf.d/captain-root.conf, /etc/nginx/conf.d/captain.conf do not look like template files, but like finalised files. 
The path /captain/generated/nginx/ mentioned in the docs does not exist. 

I appreciate any hints you may give me.

**Comment by githubsaturn** [comment, marked as answer]
- For `captain.your.domain.com` you need to make changes in the global settings (under Settings > CapRover Config Location)
- For any apps, you need to make changes in the nginx config pertaining that particular app. Go to your app > http settings > nginx config


Anywhere you see `ssl_certificate` and `ssl_certificate_key`, you'll have to update to the correct path, for example:
```
        listen              443 ssl;
        ssl_certificate     /nginx-shared/certificate;
        ssl_certificate_key /nginx-shared/certificate-key;
```

**Comment by mjmare** [comment]
Super helpful! 
Got it to work for each app. 
For some reason I missed the per app nginx config. I think I was misled by the wording "Edit default Nginx configurations".

Can I set the wildcard in one go for all apps at once?

**Reply by githubsaturn** [reply]
Not currently, but will be available in the next version:
https://github.com/caprover/caprover/pull/1377
